server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/JobCallerBuilder.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.task.caller;
+
+import java.util.Map;
+
+import com.oceanbase.odc.common.util.SystemUtils;
+import com.oceanbase.odc.service.task.constants.JobConstants;
+import com.oceanbase.odc.service.task.constants.JobEnvConstants;
+import com.oceanbase.odc.service.task.enums.TaskRunModeEnum;
+
+/**
+ * @author yaobin
+ * @date 2023-11-29
+ * @since 4.2.4
+ */
+public class JobCallerBuilder {
+
+    public static JobCaller buildJvmCaller() {
+        return new JvmJobCaller();
+    }
+
+
+    public static JobCaller buildK8sJobCaller(K8sJobClient k8sJobClient, PodConfig podConfig) {
+
+        PodParam podParam = podConfig.getPodParam();
+
+        Map<String, String> envs = podParam.getEnvironments();
+        envs.put(JobEnvConstants.BOOT_MODE, JobConstants.ODC_BOOT_MODE_EXECUTOR);
+        envs.put(JobEnvConstants.TASK_RUN_MODE, TaskRunModeEnum.K8S.name());
+
+        envs.put("DATABASE_HOST", SystemUtils.getEnvOrProperty("ODC_DATABASE_HOST"));
```

Comment:
what are these properties used for? job worker needs them? worker may not attach the metadb if it is running in an isolated env.